core/src/main/java/nu/yona/server/device/service/DeviceService.java

```diff
@@ -0,0 +1,32 @@
+package nu.yona.server.device.service;
+
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import nu.yona.server.device.entities.UserDeviceRepository;
+import nu.yona.server.subscriptions.service.UserService;
+
+@Service
+public class DeviceService
+{
+	@Autowired(required = false)
+	private UserService userService;
+
+	@Autowired(required = false)
+	private UserDeviceRepository userDeviceRepository;
+
+	public Set<DeviceBaseDto> getDevicesOfUser(UUID userId)
```

Comment:
Why not return `Set<UserDeviceDto>`?

Reply:
I'm not done with this yet. It's quite possible that it'll change that way.